cmd/skopeo/delete.go

```diff
@@ -3,44 +3,52 @@ package main
 import (
 	"errors"
 	"fmt"
-	"io"
 	"strings"
 
 	"github.com/containers/image/v5/transports"
 	"github.com/containers/image/v5/transports/alltransports"
-	"github.com/urfave/cli"
+	"github.com/spf13/cobra"
 )
 
 type deleteOptions struct {
 	global *globalOptions
 	image  *imageOptions
 }
 
-func deleteCmd(global *globalOptions) cli.Command {
-	sharedFlags, sharedOpts := sharedImageFlags()
-	imageFlags, imageOpts := imageFlags(global, sharedOpts, "", "")
-	opts := deleteOptions{
-		global: global,
-		image:  imageOpts,
-	}
-	return cli.Command{
-		Name:  "delete",
-		Usage: "Delete image IMAGE-NAME",
-		Description: fmt.Sprintf(`
-	Delete an "IMAGE_NAME" from a transport
+func deleteCommand(globalFlagResults *globalOptions) *cobra.Command {
+	var (
+		opts              deleteOptions
+		deleteDescription = fmt.Sprintf(`Delete an "IMAGE_NAME" from a transport
 
-	Supported transports:
-	%s
+Supported transports:
+%s
 
-	See skopeo(1) section "IMAGE NAMES" for the expected format
-	`, strings.Join(transports.ListNames(), ", ")),
-		ArgsUsage: "IMAGE-NAME",
-		Action:    commandAction(opts.run),
-		Flags:     append(sharedFlags, imageFlags...),
+See skopeo(1) section "IMAGE NAMES" for the expected format
+`, strings.Join(transports.ListNames(), ", "))
+	)
+	deleteCommand := &cobra.Command{
+		Use:   "delete [command options] IMAGE-NAME",
+		Short: "Delete image IMAGE-NAME",
+		Long:  deleteDescription,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return deleteCmd(cmd, args, opts)
+		},
+		Example: `skopeo delete docker://registry.example.com/example/pause:latest`,
 	}
+	deleteCommand.SetUsageTemplate(usageTemplate())
+	deleteCommand.DisableFlagsInUseLine = true
+
+	flags := deleteCommand.Flags()
+	sharedFlags, sharedOpts := sharedImageFlags()
+	imageFlags, imageOpts := imageFlags(globalFlagResults, sharedOpts, "", "")
+	opts.image = imageOpts
+	opts.global = globalFlagResults
+	flags.AddFlagSet(&sharedFlags)
+	flags.AddFlagSet(&imageFlags)
+	return deleteCommand
 }
 
-func (opts *deleteOptions) run(args []string, stdout io.Writer) error {
+func deleteCmd(c *cobra.Command, args []string, opts deleteOptions) error {
```

Comment:
Why was this method turned into a global? I don’t think we need all those names separately in the package namespace. The `deleteCmd`/`deleteCommand` distinction is rather awkward, as well as calling this function `deleteCmd` (which suggests “delete a command” much more than “perform the delete operation”).